include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp

```diff
@@ -150,6 +150,36 @@ static std::optional<unsigned> getBitWidth(Type a) {
   return std::nullopt;
 }
 
+static void processTwoStridesLastDim(RankedTensorType type, ConversionPatternRewriter &rewriter,
+    std::function<void(SmallVector<OpFoldResult, 4> &, SmallVector<OpFoldResult, 4>&, SmallVector<OpFoldResult, 4>&, int)> op) {
+  int64_t rank = type.getRank();
+  auto shape = type.getShape();
+
+  SmallVector<OpFoldResult, 4> offsets, sizes, strides;
+  std::vector<int64_t> stridesInt(rank, 0);
+
+  int64_t currentStride = 1;
+  for (size_t i = shape.size(); i > 0; --i) {
+    stridesInt[i - 1] = currentStride;
+    currentStride *= stridesInt[i - 1];
+  }
+
```

Comment:
Let me know if my understanding is correct: this loop will end up assigning 1s to all elements in `stridesInt` right? `currentStride` starts out at 1, and each step we multiply it again with stridesInt at `i` which is also `1.

Reply:
thanks for noticing, sorry, refactoring artifact, should be: `currentStride *= shape[i - 1];` so the very last one is 1, and they grow as a product of shape

Reply:
actually, after some experiments, I think this loop is redundant and all strides are 1, removed